main/application.cpp

```diff
@@ -54,6 +54,12 @@ Application::Application(int argc, char *argv[])
             parameterModel, SLOT(updateAlgParameters(QString)));
     connect(qmlRoot, SIGNAL(instantiate(QString)),
             parameterModel, SLOT(createCommand(QString)));
+    for (Algorithm* alg : parameterModel->getAlgorithmList()->getAlgs()) {
+      connect(alg, SIGNAL(log), [qmlRoot](const QString msg, const bool isError){
+        QMetaObject::invokeMethod(qmlRoot, "log", Q_ARG(QVariant, msg), Q_ARG(QVariant, isError));
+      });
```

Comment:
My compile errors all point here. AFAIK, `connect` is supposed to have four parameters: a reference to the slot object, the slot function, a reference to the signal object, and the signal function. You only seem to have three of the four (I think you're missing the signal object?) Also, it's unclear to me if the second parameter should be `SIGNAL(log)` or `SIGNAL(log(QString, bool))` to capture the parameters.

Reply:
Sorry about that, I had something different that didn't have compile errors! I tried copying the same method for hooking up the `log` signal with the corresponding slot that we have in line 102 of `application.cpp`, but it gave me build errors regarding `Q_OBJECT`. I just pushed the change to fix the compile errors, so if you try building, you should see the error that I'm talking about